Firestore/Source/Model/FSTFieldValue.h

```diff
@@ -40,6 +42,32 @@ typedef NS_ENUM(NSInteger, FSTTypeOrder) {
   FSTTypeOrderObject,
 };
 
+/** Defines the return value for pending server timestamps. */
+typedef NS_ENUM(NSInteger, FSTServerTimestampBehavior) {
+  FSTServerTimestampBehaviorDefault,
+  FSTServerTimestampBehaviorEstimate,
+  FSTServerTimestampBehaviorPrevious,
+};
+
+/** Holds properties that define field value deserialization options. */
+@interface FSTFieldValueOptions : NSObject
+
+@property(nonatomic, readonly, assign) FSTServerTimestampBehavior serverTimestampBehavior;
+
+- (instancetype)init NS_UNAVAILABLE;
+
+/**
+ * Creates a FSTFieldValueOptions instance that specifies deserialization behavior for pending
+ * server timestamps.
+ */
+- (instancetype)initWithServerTimestampBehavior:(FSTServerTimestampBehavior)serverTimestampBehavior
+    NS_DESIGNATED_INITIALIZER;
+
+/** Creates FSTFieldValueOptions from FIRSnapshotOptions. */
++ (instancetype)fieldValueOptions:(FIRSnapshotOptions *)value;
```

Comment:
The label should describe its argument (and the argument should match). Maybe 

```
+(instancetype)fieldValueOptionsForSnapshotOptions:(FIRSnapshotOptions *)snapshotOptions;
```

?

Reply:
Renamed to `optionsForSnapshotOptions` (which won since it is not quite as clunky).